disassemblers/ofrak_ghidra/ofrak_ghidra/ghidra_model.py

```diff
@@ -82,6 +88,37 @@ class OfrakGhidraMixin:
     `OfrakGhidraScript` member of the class.
     """
 
+    # Hi Ed, I know this isn't the right way to do this but I don't know how to add an attribute to a Resource.
+    ofrak_code_regions = {}
```

Comment:
I was thinking that this caching would be good to put in the `GhidraProject` view - it's the bridge between the OFRAK project and the corresponding Ghidra project, so it makes sense to expand its metadata to include the vaddr offsets.

This would mean calling the script to get Ghidra's regions in `GhidraProjectAnalyzer` where the `GhidraProject` view is populated.

Also, this cache can definitely be simplified. The current data stored and the algo run on it each time represent something like a dictionary of ranges+offset pairs, where if an address is between one of those ranges, we apply to it the offset that goes with that range. Really we just need those 3 values for each code region (start, end, vaddr addend), no views or resources necessary.

Reply:
I think that it's necessary to key on the resource because if we are trying to translate addresses in two different resources, the memory mappings will be different but the address might fall in to a segment of the previously translated resource.

This would be solved by adding the memory map info to the `GhidraProject` (assuming that only one file is analyzed in on project) but I don't know how to do that.